pkg/sfu/downtrack.go

```diff
@@ -513,7 +517,7 @@ func (d *DownTrack) SetMaxSpatialLayer(spatialLayer int32) error {
 
 	d.maxSpatialLayer.set(spatialLayer)
 
-	if d.onSubscribedLayersChanged != nil {
+	if d.enabled.get() && d.onSubscribedLayersChanged != nil {
```

Comment:
Avoiding unnecessary notifications to `streamallocator` as it will trigger reallocation and potentially cause trouble.